Slim/Control/Queries.pm

```diff
@@ -4524,7 +4526,7 @@ sub yearsQuery {
 	# get them all by default
 	my $where = {};
 
-	my ($key, $table) = ($hasAlbums || $libraryID) ? ('albums.year', 'albums') : ('id', 'years');
+	my ($key, $table) = $hasAlbums ? ('albums.year', 'albums') : ('tracks.year', 'tracks');
```

Comment:
Why the removal of the library ID?

Reply:
Because we only want the base table to be albums if $hasAlbum is true.

The code a few lines down handles the join(s) necessary if we've got Library ID.

Previously, the code would not have worked properly if `$hasAlbums=false` and `$libraryID=true`: we'd still have got album years only.